homeassistant/components/advantage_air/climate.py

```diff
@@ -101,17 +102,37 @@ def __init__(self, instance: AdvantageAirData, ac_key: str) -> None:
             HVACMode.FAN_ONLY,
             HVACMode.DRY,
         ]
-        # Set supported features and HVAC modes based on current operating mode
-        if self._ac.get(ADVANTAGE_AIR_MYAUTO_ENABLED):
-            # MyAuto
-            self._attr_supported_features |= (
-                ClimateEntityFeature.TARGET_TEMPERATURE
+
+        # Add "MyTemp" preset if available
+        if ADVANTAGE_AIR_MYTEMP_ENABLED in self._ac:
+            self._attr_preset_modes += [ADVANTAGE_AIR_MYTEMP]
+            self._attr_supported_features |= ClimateEntityFeature.PRESET_MODE
+
+        # Add "MyAuto" preset if available
+        if ADVANTAGE_AIR_MYAUTO_ENABLED in self._ac:
+            self._attr_preset_modes += [ADVANTAGE_AIR_MYAUTO]
+            self._attr_supported_features |= ClimateEntityFeature.PRESET_MODE
+
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
```

Comment:
Looks like we can just set this during init?

Reply:
No, because the preset mode can change.

This PR is to move these out of init so they are dynamic to the preset.

Reply:
Maybe add a comment here and for supported features to avoid future review comments

Reply:
Comments added

Reply:
Read it wrong clearly so good to add a comment to it.
I think in general it would have looked cleaner if you would use `_handle_coordinator_update()` to update the attributes instead of all these ifs within the properties but I guess it's a matter of taste

Reply:
TIL `_handle_coordinator_update` is a thing. Ill try use that where appropriate in the future

Reply:
@gjohansson-ST I have ended up refactoring to use `_handle_coordinator_update` as you suggested, and have started using this elsewhere. Much nicer way of handling things sometimes.